### PR TITLE
fix sound on web when tab switched or browser minimized

### PR DESCRIPTION
**Pull request by jice-nospam**
This fix pauses the sound driver when the tab goes out of visibility.

### CODE REVIEW

File: uni-snd/src/web_snd.rs [hunk 1]
```diff
@@ -39,12 +52,35 @@ impl<T> SoundDriver<T> {
             err,
         }
     }
+    // -1 => game paused
+    // >0 => pause duration
+    fn get_pause_status(&self) -> f64 {
```
Comment (edwin0cheng) [inline uni-snd/src/web_snd.rs]: Maybe using Option here ? e.g
```
fn get_pause_duration(&self) -> Option<f64>
```

Comment (jice-nospam) [reply uni-snd/src/web_snd.rs]: indeed... I'm still not used to idiomatic rust :)

**Comment by jice-nospam** [comment]
There are actually 3 statuses :
* get_pause_status() == None => the game runs normally
* get_pause_status() == -1 => the game is paused (the user minimized the browser or switch to another tab) => we should stop processing sound
* get_pause_status() > 0 => the game has just been resumed. We should advance the internal time counter so that the sound resumes where it was paused and doesn't take the time elapsed while paused into account.

Maybe I can use some enum here
```
enum GameStatus {
   Running,
   Paused,
   Resumed(f32)
}
```

**Comment by jice-nospam** [comment]
done!

**Comment by edwin0cheng** [comment]
Could you mind to rebase from master and commit again ?

**Comment by jice-nospam** [comment]
rebased!

**Comment by edwin0cheng** [comment]
Merged!

**Pull request merged**